src/components/stone-handler.ts

```diff
@@ -292,4 +315,86 @@ export default class StoneHandler extends EventManager {
     // to play the audio parrallely.
     this.correctStoneAudio.play();
   }
+
+  resetStonePosition(
+    width,
+    pickedStone,
+    pickedStoneObject
+  ) {
+    const stone = pickedStone;
```

Comment:
In resetStonePosition, you are assigning const stone = pickedStone; and const stoneObj = pickedStoneObject;. Instead, directly use pickedStone and pickedStoneObject unless there’s a specific reason to reassign them.